pkg/identityserver/picture.go

```diff
@@ -97,3 +99,50 @@ func (is *IdentityServer) processUserProfilePicture(ctx context.Context, usr *tt
 
 	return
 }
+
+func (is *IdentityServer) processEndDevicePicture(ctx context.Context, dev *ttnpb.EndDevice) (err error) {
+	// External pictures, consider only largest.
+	if dev.Picture.Sizes != nil {
+		original := dev.Picture.Sizes[0]
+		if original == "" {
+			var max uint32
+			for size, url := range dev.Picture.Sizes {
+				if size > max {
+					max = size
+					original = url
+				}
+			}
+		}
+		if original != "" {
+			dev.Picture.Sizes = map[uint32]string{0: original}
+		} else {
+			dev.Picture.Sizes = nil
+		}
+	}
+
+	// Embedded (uploaded) picture. Make square.
```

Comment:
@mjamescompton are you aware of the fact that these images will be square?

Reply:
Should not be a problem, let's only change it if Wilhelmsen ask about it.